### PR TITLE
SAK-45919 Assignments dates appear in the user preference language instead of site language

### PR DESCRIPTION
**Pull request by jesusmmp**
https://sakaiproject.atlassian.net/browse/SAK-45919

Important regression from: https://github.com/sakaiproject/sakai/pull/9401

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/time/impl/UserTimeServiceImpl.java [hunk 1]
```diff
@@ -169,10 +173,9 @@ public String dateTimeFormat(Instant date, FormatStyle dateStyle, FormatStyle ti
         if (date == null) return "";
         if (dateStyle == null) { dateStyle = FormatStyle.MEDIUM; }
         if (timeStyle == null) { timeStyle = FormatStyle.SHORT; }
-        Locale locale = preferencesService.getLocale(sessionManager.getCurrentSessionUserId());
```
Comment (ottenhoff) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/time/impl/UserTimeServiceImpl.java:172]: What happens when there is no site locale set? And the user has a preferred locale?

Comment (jesusmmp) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/time/impl/UserTimeServiceImpl.java:172]: Then, dates are shown on user preferred locale language.

Comment (ottenhoff) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/time/impl/UserTimeServiceImpl.java:172]: Great. Where does that logic happen?

Comment (jesusmmp) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/time/impl/UserTimeServiceImpl.java:172]: On the next line: .withLocale(resourceLoader.getLocale());

https://github.com/sakaiproject/sakai/pull/10187/files#diff-c381b6340f804a20160ea067d2a476a4eaa3b7a1b369347fd9cf8b8d970a99baR178

**Review by ottenhoff — APPROVED** [review]

**Review by ern — APPROVED** [review]
This looks good

**Pull request merged**